### ISSUE TITLE
[bug]: Critical failure when grouping objects (Edgeless Mode)

### ISSUE THREAD

**Issue by unowhere** [toeverything/AFFiNE#172]
### Describe the bug

When I use board mode, create few text blocks, and group them, the app will break down every time.

### To Reproduce

(use on Windows 10)

1. switch to “Board mode” 
2. create few text blocks
3. click “group”
4. result as : 

https://user-images.githubusercontent.com/14874414/183808108-fd5cdbd3-60f6-4749-9d49-7e23be346e90.mp4

We've encountered an error!
A shape (affineOCRdSQkxk20I27mi) has a parent (8ea74ace-a84a-4bec-083d-ec3df822e00b) that does not exist!
Copy ErrorRefresh Page
Keep getting this error? Download your project as a backup and then reset the document.

### Screenshots

![image](https://user-images.githubusercontent.com/14874414/183808259-d73d559f-c56c-42f8-ab4b-64d2c7889516.png)

### Expected behavior

resolve the problem.

### Platform

Windows 10

### Browser

Chrome

**Comment by alt1o** [comment]
Got it. Working on it.

**Comment by icese7en** [comment]
I found that the problem is in `libs/components/affine-board/src/hooks/use-shapes.ts` which was used by `libs/components/affine-board/src/Board.tsx`, for the `setBlocks` function of `libs/components/affine-board/src/hooks/use-shapes.ts` was not triggered, so, the `blocks` were old, but their `shapeProps` was somehow updated by `group`  function of `libs/components/board-state/src/tldraw-app.ts`. 
In conclusion, the shapes were set by `libs/components/board-state/src/tldraw-app.ts`, then `libs/components/affine-board/src/hooks/use-shapes.ts` used the old `blocks` with new props to overwrite the right shapes which had been set by `libs/components/board-state/src/tldraw-app.ts`.
